src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -695,6 +697,26 @@ public function appendToFile(string $filename, $content)
         }
     }
 
+    /**
+     * Watches a file or directory for any changes, and calls $callback when any changes are detected.
+     *
+     * @param mixed    $path     The path to watch for changes. Can be a path to a file or directory, iterator or array with paths
+     * @param callable $callback The callback to execute when a change is detected
+     * @param float    $timeout  The idle timeout in milliseconds after which the process will be aborted if there are no changes detected
+     *
+     * @throws \InvalidArgumentException|IOException
+     */
+    public function watch($path, callable $callback, float $timeout = null)
+    {
+        if (\extension_loaded('inotify')) {
+            $watcher = new INotifyWatcher();
+        } else {
+            $watcher = new FileChangeWatcher();
+        }
```

Comment:
Cache the watcher instance?
```suggestion
        if (null === $this->watcher) {
            if (\extension_loaded('inotify')) {
                $this->watcher = new INotifyWatcher();
            } else {
                $this->watcher = new FileChangeWatcher();
            }
        }
```

Reply:
I'm not sure what the benefit would be here? The performance impact of checking if the extension exists and instantiating the class is negligible. And this would only be useful when running the watch function multiple times, which I don't think is a common use case.
In most cases, you would add a watch for a specific directory, which will block until you explicitly terminate the watch (by returning `false` or by setting a timeout which then expires). So I don't think it would even make sense to call the watch method multiple times